From 7fdeba2aaa38cb5f065fc98f3477747be36997cc Mon Sep 17 00:00:00 2001
From: Gabor Szadovszky <gabor.szadovszky@cloudera.com>
Date: Wed, 5 Oct 2016 13:21:40 -0700
Subject: [PATCH] =?UTF-8?q?PARQUET-685=20-=20Deprecated=20ParquetInputSpli?=
 =?UTF-8?q?t=20constructor=20passes=20paramet=E2=80=A6?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The problem was not discovered because the test was bugous. Updated both sides.

Author: Gabor Szadovszky <gabor.szadovszky@cloudera.com>

Closes #372 from gszadovszky/PARQUET-685 and squashes the following commits:

9cbeee2 [Gabor Szadovszky] PARQUET-685 - Deprecated ParquetInputSplit constructor passes parameters in the wrong order.
---
 .../java/org/apache/parquet/hadoop/ParquetInputSplit.java    | 3 +--
 .../test/java/org/apache/parquet/hadoop/TestInputFormat.java | 5 +----
 2 files changed, 2 insertions(+), 6 deletions(-)

diff --git a/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetInputSplit.java b/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetInputSplit.java
index 8a13a3cc8a..b97daa582c 100644
--- a/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetInputSplit.java
+++ b/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetInputSplit.java
@@ -26,7 +26,6 @@
 import java.io.DataOutputStream;
 import java.io.IOException;
 import java.util.Arrays;
-import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.zip.GZIPInputStream;
@@ -89,7 +88,7 @@ public ParquetInputSplit(
       String fileSchema,
       Map<String, String> extraMetadata,
       Map<String, String> readSupportMetadata) {
-    this(path, start, length, end(blocks, requestedSchema), hosts, offsets(blocks));
+    this(path, start, end(blocks, requestedSchema), length, hosts, offsets(blocks));
   }
 
   private static long end(List<BlockMetaData> blocks, String requestedSchema) {
diff --git a/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestInputFormat.java b/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestInputFormat.java
index 9fe30080ef..d8b3de3d36 100644
--- a/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestInputFormat.java
+++ b/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestInputFormat.java
@@ -481,11 +481,8 @@ private List<ParquetInputSplit> generateSplitByDeprecatedConstructor(long min, l
         .generateSplitInfo(blocks, hdfsBlocks, min, max);
 
     for (ClientSideMetadataSplitStrategy.SplitInfo splitInfo : splitInfos) {
-      BlockMetaData lastRowGroup = splitInfo.getRowGroups().get(splitInfo.getRowGroupCount() - 1);
-      long end = lastRowGroup.getStartingPos() + lastRowGroup.getTotalByteSize();
-
       ParquetInputSplit split = new ParquetInputSplit(fileStatus.getPath(),
-          splitInfo.hdfsBlock.getOffset(), end, splitInfo.hdfsBlock.getHosts(),
+          splitInfo.hdfsBlock.getOffset(), splitInfo.hdfsBlock.getLength(), splitInfo.hdfsBlock.getHosts(),
           splitInfo.rowGroups, schema.toString(), null, null, extramd);
       splits.add(split);
     }
